packages/astro-embed-baseline-status/BaselineStatus.astro

```diff
@@ -0,0 +1,293 @@
+		chrome = edge = firefox = safari = 'unknown';
+	}
+	return `Baseline: ${title}${year ? ` ${year}` : ''}${
+		newlyAvailable ? ` (newly available)` : ''
+	}. Supported in Chrome: ${
+		chrome === 'available' ? 'yes' : chrome
+	}. Supported in Edge: ${
+		edge === 'available' ? 'yes' : edge
+	}. Supported in Firefox: ${
+		firefox === 'available' ? 'yes' : firefox
+	}. Supported in Safari: ${safari === 'available' ? 'yes' : safari}.`;
+}
+---
+
+<baseline-status>
+	<h1>{feature.name}</h1>
+	<details>
+		<summary
+			aria-label={getAriaLabel(
+				title,
+				year,
+				baseline === 'newly',
+				chrome?.status,
+				edge?.status,
+				firefox?.status,
+				safari?.status
+			)}
+		>
+			<BaselineIcon support={baseline} />
+			<div class="baseline-status-title" aria-hidden="true">
+				<div>
+					{
+						!(baseline === 'limited' || baseline === 'no_data') && (
+							<strong>Baseline</strong>
+						)
+					}
+					{title}
+					{year}
+					{
+						baseline === 'newly' && (
+							<span class="baseline-badge">newly available</span>
+						)
+					}
+				</div>
+				<div class="baseline-status-browsers">
+					<span>
+						<BrowserIcon browser="chrome" />
+						<SupportIcon {baseline} browserImplementation={chrome} />
+					</span>
+					<span>
+						<BrowserIcon browser="edge" />
+						<SupportIcon {baseline} browserImplementation={edge} />
+					</span>
+					<span>
+						<BrowserIcon browser="firefox" />
+						<SupportIcon {baseline} browserImplementation={firefox} />
+					</span>
+					<span>
+						<BrowserIcon browser="safari" />
+						<SupportIcon {baseline} browserImplementation={safari} />
+					</span>
+				</div>
+			</div>
+			<span class="open-icon" aria-hidden="true">
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					width="11"
+					height="7"
+					viewBox="0 0 11 7"
+					fill="none"
+				>
+					<path
+						d="M5.5 6.45356L0.25 1.20356L1.19063 0.262939L5.5 4.59419L9.80937 0.284814L10.75 1.22544L5.5 6.45356Z"
+						fill="currentColor"></path>
+				</svg>
+			</span>
+		</summary>
+		<p set:html={description} />
+		<p
+			set:html={baseline === 'no_data'
+				? ''
+				: `<a href="https://github.com/web-platform-dx/web-features/blob/main/features/${feature.feature_id}.yml" target="_top">Learn more</a>`}
+		/>
+	</details>
+</baseline-status>
+
+<style>
+	baseline-status {
+		--baseline-status-color-limited: light-dark(#ea8600, #f09418);
+		--baseline-status-color-newly: light-dark(#1a73e8, #4185ff);
+		--baseline-status-color-widely: light-dark(#1e8e3e, #24a446);
+		--baseline-status-color-no_data: light-dark(#707070, #868686);
+		--baseline-status-color-outline: light-dark(#d9d9d9, #d9d9d9);
+		--baseline-status-color-text: light-dark(#000, #fff);
+		--baseline-status-color-link: light-dark(#1a73e8, #5aa1ff);
+
+		color: var(--baseline-status-color-text);
+		display: block;
+		border: solid 1px var(--baseline-status-color-outline);
+		border-radius: 8px;
+		padding: 16px 24px 0 24px;
+		max-width: 800px;
```

Comment:
It would be nice to use logical properties instead of physical ones.

If this sounds interesting, I have a handy dandy tool that you could use: https://logical.mynk.app/